src/Microsoft.AspNetCore.Mvc.Core/AcceptedResult.cs

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// An <see cref="ActionResult"/> that returns an Accepted (202) response with a Location header.
+    /// </summary>
+    public class AcceptedResult : ObjectResult
+    {
+        private string _location;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AcceptedResult"/> class with the values
+        /// provided.
+        /// </summary>
+        /// <param name="location">The location at which the status of requested content can be monitored.</param>
+        /// <param name="value">The value to format in the entity body.</param>
+        public AcceptedResult(string location, object value)
+            : base(value)
+        {
+            if (location == null)
```

Comment:
We should remove the null check here. `location` would be an optional component.